ext/ffi/ffi.c

```diff
@@ -746,6 +762,36 @@ static zend_always_inline zend_result zend_ffi_zval_to_cdata(void *ptr, zend_ffi
 		zend_ffi_cdata *cdata = (zend_ffi_cdata*)Z_OBJ_P(value);
 		if (zend_ffi_is_compatible_type(type, ZEND_FFI_TYPE(cdata->type)) &&
 			type->size == ZEND_FFI_TYPE(cdata->type)->size) {
+			if (is_ret && type->size < sizeof(ffi_arg)) {
```

Comment:
This is an ugly solution but I don't see a better way so far.
Also, perhaps it is better to put this inside an ifdef for big endian to reduce code bloat but I'm not sure.